### ISSUE TITLE
Volontär kan boka lediga volontäruppdrag på alla boenden

### ISSUE THREAD

**Issue by FannyBacklin** [noQ-sweden/noq#646]
**Story**
Som en volontär vill jag kunna boka lediga volontäruppdrag på samtliga boenden, så att jag enkelt kan välja och anmäla mig till uppdrag som passar mig och mitt schema.

Ref to EPIC: Ny roll - Volontär [#509](https://github.com/noQ-sweden/noq/issues/509)

**Acceptance Criteria (checklist)**
- [ ] Volontären ska kunna klicka på ett ledigt uppdrag och välja att boka det.
- [ ] Vid bokning ska systemet markera uppdraget som bokat för den specifika volontären.
- [ ] Efter att ett uppdrag har bokats ska volontären få en bekräftelse som innehåller alla relevanta uppdragsdetaljer (boende, datum, tid, plats, uppdragstitel).
- [ ] Volontären ska även kunna se sina bokade uppdrag i en egen översikt.
- [ ] Volontären ska kunna avboka ett uppdrag om det behövs.
- [ ] Vid avbokning ska uppdraget automatiskt bli tillgängligt för andra volontärer.

**Acceptance Criteria (BDD)**

AC1:
Givet att jag ser ett ledigt uppdrag i listan,
När jag klickar på "Boka" för det uppdraget,
Så ska jag kunna bekräfta min bokning.

AC2:
Givet att jag har bekräftat min bokning,
När bokningen har genomförts,
Så ska systemet markera uppdraget som bokat för min profil,
Och det ska inte längre visas som tillgängligt för andra volontärer.

AC3:
Givet att jag har bokat ett uppdrag,
När bokningen är genomförd,
Så ska jag få en bekräftelse som innehåller:
Boendets namn,
Datum,
Tid,
Plats,
Uppdragets titel.

AC4:
Givet att jag har bokat ett uppdrag,
När jag väljer att avboka uppdraget,
Så ska uppdraget tas bort från min översikt,
Och det ska åter bli tillgängligt för andra volontärer.

AC5:
Givet att jag har avbokat ett uppdrag,
När avbokningen har genomförts,
Så ska det uppdraget återigen visas som tillgängligt i listan över volontäruppdrag.

**Comment by FannyBacklin** [comment]
@overath2 Kan vi ha avboka i samma story eller bör de säras på?

**Comment by overath2** [comment]
En user story kan ha flera acceptanskriteria men akta så den inte blir för stor bara.

**Comment by patrik-pwned** [comment]
Jag funderar om uppdragsgivaren bör kunna godkänna/neka en bokning av ett åtagande. Alternativt blocka vissa volontärer. Jag föreställer mig att en volontär kanske inte var så bra första gången och att uppdragsgivaren inte vill ha den volontären en andra gång. Det är kanske ett senare problem.

**Comment by overath2** [comment]
Sådan här bra idéer och frågor poppar upp i POs huvud hela tiden. Fundera hur du vill göra. Lägga in dem i GitHub eller spara i en fil med  idéer så spammar du inte. Var också tydlig med om du vill ha idéer från andra eller ej? I detta fall skulle jag säga att funktionen eventuell bli aktuell efter nyår.

**Comment by overath2** [comment]
Tänk bråttom och viktigt som två parametrar för din ranking. Googla eisenhover matrix så fattar du.

**Comment by patrik-pwned** [comment]
@FannyBacklin hur gör vi med denna?
(frågar för att den inte lämnat backlogen) :)

**Comment by patrik-pwned** [comment]
Tycker att vi bordlägger detta till efter årsskiftet. 

Ge en tumma upp för OK och en tumme ner för ha med i sprint 11/12

**Comment by overath2** [comment]
Ett måste för lansering och bör finnas i MVP.

**Comment by patrik-pwned** [comment]
Är det noQ eller de olika boenden som kommer att lägga upp lediga uppdrag?

Tyckte att Malin sa att de inte använder sig eller tar in volontärer på akut- resp stödboenden.

**Comment by overath2** [comment]
fel av mig. Läste fel.
Denna är inte ett måste för lansering.
Här pratar vi om volontär. Jag tänkte stödperson.
Stödperson måste kunna boka vid lansering.
Stödperson är en (1) typ av volontär.
Volontärer fixar vi senare.
Rörigt...

**Comment by overath2** [comment]
This huge story has a lot of great stuff within. Have this been captured elsewhere @FannyBacklin and @patrik-pwned ?

**Comment by patrik-pwned** [comment]
Yes and no. I have drawn a lot of inspiration from this ticket and some are or have been worked on but I let this ticket lay dormant somewhere in the backlog so we wouldn't lose any good ideas for when we need more ideas on what to improve :)
